src/ResourceManager/Consumption/Commands.Consumption/Cmdlets/Budget/NewAzureRmConsumptionBudget.cs

```diff
@@ -0,0 +1,243 @@
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Cost", "Usage")]
+        public string Category;        
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = true, HelpMessage = HelpMessages.TimeGrain)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = true, HelpMessage = HelpMessages.TimeGrain)]
+        [ValidateNotNullOrEmpty]
+        [ValidateSet("Monthly", "Quarterly", "Annually")]
+        public string TimeGrain;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = true, HelpMessage = HelpMessages.StartDate)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = true, HelpMessage = HelpMessages.StartDate)]
+        [ValidateNotNullOrEmpty]
+        public DateTime? StartDate;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.EndDate)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.EndDate)]
+        [ValidateNotNullOrEmpty]
+        public DateTime? EndDate;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupName)]
+        [ValidateNotNullOrEmpty]
+        [ResourceGroupCompleter]
+        public string ResourceGroupName;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.MeterFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.MeterFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] MeterFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] ResourceFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupFilter)]
+        [Parameter(ParameterSetName = ParameterSetNames.NotificationItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.ResourceGroupFilter)]
+        [ValidateNotNullOrEmpty]
+        public string[] ResourceGroupFilter;
+
+        [Parameter(ParameterSetName = ParameterSetNames.SubscriptionItemParameterSet, Mandatory = false, HelpMessage = HelpMessages.NotificationKey)]
```

Comment:
I realized that I was probably not explicit enough about this.  To do the parameter sets properly, please remove all of the notification related parameters (basically any that require notification key to work correctly) from the SubscriptionItemParameterSet.